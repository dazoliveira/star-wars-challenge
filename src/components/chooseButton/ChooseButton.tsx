import { Button, Grid } from "@material-ui/core";
import { useStyles } from "./styles";

export type ChooseButtonProps = {
    handleOnClick(): void
    isMobile: boolean
    variant: boolean
    text: string
    isDisabled: boolean
}

export default function ChooseButton({ 
  handleOnClick, 
  isMobile, 
  variant,
  text,
  isDisabled
}: ChooseButtonProps) {

    const classes = useStyles({ variant,  isMobile})
    
    return (    
      <Grid 
        item xs={12} 
        style={{ textAlign: 'center' }}
      >
          <Button 
            data-testid='choose-button'
            className={classes.root}
            onClick={handleOnClick}
            disabled={isDisabled} 
            variant="contained"
          >
            {text}
          </Button>
      </Grid>   
    )
}