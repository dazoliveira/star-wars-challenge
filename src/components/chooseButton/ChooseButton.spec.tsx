import { screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event';
import { renderWithTheme } from "utils/tests/helpers";
import ChooseButton, { ChooseButtonProps } from './ChooseButton'

const props: ChooseButtonProps = {
    handleOnClick: jest.fn(),
    isMobile: false,
    variant: true,
    text: 'choose your path again, Padawan',
    isDisabled: false
}

describe('<ChooseButton />', () => {
    it('should render choose button able', () => {
        const { container } = renderWithTheme(<ChooseButton {...props} />)

        expect(screen.queryByTestId('choose-button')).toBeInTheDocument()
        expect(screen.queryByTestId('choose-button')).not.toHaveClass("Mui-disabled")

        expect(container.firstChild).toMatchSnapshot()
    })

    it('should call choose button click handle', async () => {

        renderWithTheme(<ChooseButton {...props }/>)

        const chooseButton = screen.getByRole('button', {
            name: 'choose your path again, Padawan'
        })

        await userEvent.click(chooseButton)

        expect(props.handleOnClick).toHaveBeenCalled()
    })

    it('should render disable button', () => {

        props.isDisabled = true

        const {getByTestId} = renderWithTheme(<ChooseButton {...props }/>)

        const chooseButton = getByTestId('choose-button')
  
        expect(chooseButton).toHaveClass("Mui-disabled") 
    })

    it('should render correct button text', () => {
        const { getByText } = renderWithTheme(<ChooseButton {...props }/>)

        const text = getByText('choose your path again, Padawan')
  
        expect(text).toBeInTheDocument() 
    })
});