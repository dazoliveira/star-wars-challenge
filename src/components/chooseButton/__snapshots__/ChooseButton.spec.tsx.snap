// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseButton /> should render choose button able 1`] = `
<div
  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  style="text-align: center;"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-root-1 makeStyles-root-2"
    data-testid="choose-button"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      choose your path again, Padawan
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</div>
`;
